scripts/msct_moco.py

```diff
@@ -66,160 +68,242 @@ def moco(param):
     sct.create_folder(folder_mat)
 
     # Get size of data
-    sct.printv('\nGet dimensions data...', verbose)
-    data_im = Image(file_data + ext)
-    nx, ny, nz, nt, px, py, pz, pt = data_im.dim
-    sct.printv(('.. ' + str(nx) + ' x ' + str(ny) + ' x ' + str(nz) + ' x ' + str(nt)), verbose)
+    sct.printv('\nData dimensions:', verbose)
+    im_data = Image(file_data + ext)
+    nx, ny, nz, nt, px, py, pz, pt = im_data.dim
+    sct.printv(('  ' + str(nx) + ' x ' + str(ny) + ' x ' + str(nz) + ' x ' + str(nt)), verbose)
 
     # copy file_target to a temporary file
     sct.printv('\nCopy file_target to a temporary file...', verbose)
     sct.copy(file_target + ext, 'target.nii')
     file_target = 'target'
+    if not param.fname_mask == '':
+        file_mask = 'mask.nii'
+        convert(param.fname_mask, file_mask, squeeze_data=False)
+        im_maskz_list = [file_mask]  # use a list with single element if not sagittal. Otherwise, will be updated.
+
+        # If scan is sagittal, split src and target along Z (slice)
+    if param.is_sagittal:
+        dim_sag = 2  # TODO: find it
+        # z-split data (time series)
+        im_z_list = split_data(im_data, dim=dim_sag, squeeze_data=False)
+        file_data_splitZ = []
+        for im_z in im_z_list:
+            im_z.save()
+            file_data_splitZ.append(im_z.absolutepath)
+        # z-split target
+        im_targetz_list = split_data(Image(file_target+ext), dim=dim_sag, squeeze_data=False)
+        file_target_splitZ = []
+        for im_targetz in im_targetz_list:
+            im_targetz.save()
+            file_target_splitZ.append(im_targetz.absolutepath)
+        # z-split mask (if exists)
+        if not param.fname_mask == '':
+            im_maskz_list = split_data(Image(file_mask), dim=dim_sag, squeeze_data=False)
+            file_mask_splitZ = []
+            for im_maskz in im_maskz_list:
+                im_maskz.save()
+                file_mask_splitZ.append(im_maskz.absolutepath)
+        # initialize file list for output matrices
+        file_mat = np.chararray([nz, nt],
```

Comment:
`chararray` is deprecated and also not a good idea
https://docs.scipy.org/doc/numpy/reference/generated/numpy.chararray.html

Reply:
after spending 10min on this i need help-- i've tried 
~~~
file_mat = np.array([nz, nt], dtype="str")
~~~
but pb because string is immutable. solution?